### PR TITLE
 python3Packages.xdis: 4.0.1 -> 4.0.3 

### PR DESCRIPTION
**Pull request by jonringer**
###### Motivation for this change
noticed it was broken while reviewing a few other packages

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @

### CODE REVIEW

**Comment by jonringer** [comment]
Upstreamed issue for python2.7 here https://github.com/rocky/python-xdis/issues/51

**Comment by jonringer** [comment]
Wow, maintainer fixed the tests already, now just using fetchpatch until next version bump.

File: pkgs/development/python-modules/xdis/default.nix [hunk 1]
```diff
@@ -1,28 +1,35 @@
...
+        sha256 = "1vdxvwgw9w5rj6n6i6dyqancx7q7rm21k1ys7rq6glpyc9r6abc8";
+    })
+  ];
+
   checkInputs = [ pytest ];
   propagatedBuildInputs = [ six click ];
 
   checkPhase = ''
-    make check
+    pytest
```
Comment (lsix) [inline pkgs/development/python-modules/xdis/default.nix]: Why did you change this ?

According to https://github.com/rocky/python-xdis/blob/master/README.rst#testing `make check` seems to be the recommended way to execute the tests. I think we should prefer to do what is advised by upstream, unless there is a rational not too do so.

Comment (jonringer) [reply pkgs/development/python-modules/xdis/default.nix]: because the pypi package no longer ships the `pytest` directory. See https://github.com/rocky/python-xdis/issues/51

Comment (lsix) [reply pkgs/development/python-modules/xdis/default.nix]: In such situations, I think the preferred usage is to fetch the source directly from github, which includes tests:

```patch
diff --git a/pkgs/development/python-modules/xdis/default.nix b/pkgs/development/python-modules/xdis/default.nix
index ea6f01d1a6d..4657940cbf2 100644
--- a/pkgs/development/python-modules/xdis/default.nix
+++ b/pkgs/development/python-modules/xdis/default.nix
@@ -1,4 +1,4 @@
-{ lib, buildPythonPackage, fetchPypi, fetchpatch
+{ lib, buildPythonPackage, fetchFromGitHub, fetchpatch
 , click
 , pytest
 , six
@@ -8,9 +8,11 @@ buildPythonPackage rec {
   pname = "xdis";
   version = "4.0.3";
 
-  src = fetchPypi {
-    inherit pname version;
-    sha256 = "09fsvvh2alnbs3gv9pkx9ch4x4p0i156518hn9kg908gizgzblk0";
+  src = fetchFromGitHub {
+    owner = "rocky";
+    repo = "python-xdis";
+    rev = "${version}";
+    sha256 = "1h4j8hincf49zyd0rvn4bh0ypj8836y8vz3d496ycb9gjzkr6044";
   };
 
   patches = [
@@ -26,7 +28,7 @@ buildPythonPackage rec {
   propagatedBuildInputs = [ six click ];
 
   checkPhase = ''
-    pytest
+    make check
   '';
 
   meta = with lib; {
```

Comment (jonringer) [reply pkgs/development/python-modules/xdis/default.nix]: both of them have tests, the pytest directory (from my understanding) was to just help with setup.

Comment (jonringer) [reply pkgs/development/python-modules/xdis/default.nix]: i couldn't make a ticket about failing tests, if there were no test to be ran ;)

Comment (lsix) [reply pkgs/development/python-modules/xdis/default.nix]: Yes sure, I do entirely realize that there are some tests outside, but clearly, some tests are located in the `pytest` directory:

```
$ tree pytest/
pytest/
├── Makefile
├── test_bytecode.py
├── test_disasm.py
├── test_instructions.py
├── test_load_file.py
├── test_opcode.py
├── test_std.py
├── testdata
│   ├── big_dict-3.6.right
│   ├── fstring-3.6.right
│   ├── multi-fn-2.7.pyc
│   ├── multi-fn-3.6.pyc
│   ├── multi-fn.py
│   ├── pypy_lambda.right
│   └── raise-3.0.right
└── todo-test.py

1 directory, 15 files
```

As far as I can see, there are tests in `test`, `pytest` and probably some in `test_unit`

Comment (jonringer) [reply pkgs/development/python-modules/xdis/default.nix]: hmm, you're right. I'll change.

**Comment by jonringer** [comment]
patch was used to fix the tests when only running `pytest`, since I'm checking out from the repo now, it's unnecessary and the default `make check` works now.

**Comment by lsix** [comment]
That's odd, but ok.

I can run `nix-review`, everything builds fine.

Thanks for the great work!

I’ll merge shortly
